Sources/MapLibreSwiftUI/Models/MapCamera/MapViewCamera.swift

```diff
@@ -32,25 +47,76 @@ public struct MapViewCamera {
     /// - Returns: The constructed MapViewCamera.
     public static func center(_ coordinate: CLLocationCoordinate2D,
                               zoom: Double,
-                              pitch: Double = 90.0,
-                              course: Double = 0) -> MapViewCamera {
+                              pitch: Double? = Defaults.pitch,
+                              course: Double = Defaults.course,
+                              reason: CameraChangeReason? = nil) -> MapViewCamera {
         
         return MapViewCamera(state: .centered,
                              coordinate: coordinate,
                              zoom: zoom,
                              pitch: pitch,
-                             course: course)
+                             course: course,
+                             lastReasonForChange: reason)
     }
     
-    public static func trackUserLocation(_ location: CLLocation,
-                                         zoom: Double,
-                                         pitch: Double = 90.0) -> MapViewCamera {
+    /// Enables user location tracking within the MapView.
+    ///
+    /// This feature uses the MLNMapView's userTrackingMode = .follow
+    ///
+    /// - Parameters:
+    ///   - zoom: Set the desired zoom. This is a one time event and the user can manipulate their zoom after unlike pitch.
+    ///   - pitch: Provide a fixed pitch value. The user will not be able to adjust pitch using gestures when this is set. Use nil/default to allow user control.
+    /// - Returns: The MapViewCamera representing the scenario
+    public static func trackUserLocation(zoom: Double = Defaults.zoom,
+                                         pitch: Double? = Defaults.pitch) -> MapViewCamera {
         
+        // Coordinate is ignored when tracking user location. However, pitch and zoom are valid.
         return MapViewCamera(state: .trackingUserLocation,
-                             coordinate: location.coordinate,
+                             coordinate: Defaults.coordinate,
```

Comment:
Hmm... Let's make illegal state (or actions taken due to misunderstanding of magic values) impossible at the type layer.

I think Swift has pretty rich enum support, so we should actually model these "maybe properties" inside the variants.

Concretely, I think the enum should look something more like the following:

```swift
public enum CameraState: Hashable, Codable {

    /// Centered on a coordinate
    case centered(centerCoordinate: CLLocationCoordinate2D)
    
    /// Follow the user's location using the MapView's internal camera.
    case trackingUserLocation

    /// And so on.... This also lets us model things like rect more richly if desired
}
```